Marlin/src/sd/cardreader.cpp

```diff
@@ -560,10 +560,11 @@ void CardReader::checkautostart() {
   if (autostart_index < 0 || flag.sdprinting) return;
 
   if (!isDetected()) initsd();
-
-  #if ENABLED(EEPROM_SETTINGS) && DISABLED(FLASH_EEPROM_EMULATION)
-    SERIAL_ECHOLNPGM("Loading settings from SD");
-    (void)settings.load();
+  #if ENABLED(EEPROM_SETTINGS) && NONE(FLASH_EEPROM_EMULATION, SPI_EEPROM, I2C_EEPROM)
```

Comment:
This condition means it could be either a "native" EEPROM or an SD EEPROM. If it is a "native" EEPROM then we don't need to load settings here. Is it possible to add a `NATIVE_EEPROM` flag to all the platform that we can confirm have a "native" EEPROM (i.e., An EEPROM that is written using an external library and which uses neither SPI nor i2c for the interface.)

Reply:
else you can do a local define in this file...
#define SD_EEPROM ENABLED(EEPROM_SETTINGS) && NONE(FLASH_EEPROM_EMULATION, SPI_EEPROM, I2C_EEPROM)

but i prefer my first trivial implementation, which allow to query the loaded status... the SD can be put later after the boot...

Reply:
But wouldn't that cause `SD_EEPROM` to be true when using a Mega2560 with built-in EEPROM?